src/layout/List/ListComponent.tsx

```diff
@@ -74,45 +83,122 @@ export const ListComponent = ({ node }: IListProps) => {
+    return (formData?.saveToList as Row[]).some((selectedRow) =>
+      Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]),
+    );
+  }
+
   const title = item.textResourceBindings?.title;
   const description = item.textResourceBindings?.description;
+
+  const handleRowClick = (row) => {
+    if (saveToList) {
+      handleSelectedCheckboxRow(row);
+    } else {
+      handleSelectedRadioRow({ selectedValue: row });
+    }
+  };
+
+  const handleSelectedCheckboxRow = (row) => {
+    if (!saveToList) {
+      return;
+    }
+    if (isRowChecked(row)) {
+      const index = (formData?.saveToList as Row[]).findIndex((selectedRow) => {
+        const { altinnRowId: _ } = selectedRow;
+        return Object.keys(row).every((key) => Object.hasOwn(selectedRow, key) && row[key] === selectedRow[key]);
+      });
+      if (index >= 0) {
+        removeFromList({
+          reference: saveToList,
+          index,
+        });
+      }
+    } else {
+      const uuid = uuidv4();
+      const next: Row = { [ALTINN_ROW_ID]: uuid };
+      for (const binding of Object.keys(bindings)) {
+        if (binding != 'saveToList') {
+          next[binding] = row[binding];
+        }
+      }
+      appendToList({
+        reference: saveToList,
+        newValue: { ...next },
+      });
+    }
+  };
+
+  const renderListItems = (row, tableHeaders) =>
+    tableHeadersToShowInMobile.map((key) => (
+      <div key={key}>
+        <strong>
+          <Lang id={tableHeaders[key]} />
+        </strong>
+        <span>{typeof row[key] === 'string' ? <Lang id={row[key]} /> : row[key]}</span>
+      </div>
+    ));
+
   if (isMobile) {
     return (
       <ComponentStructureWrapper node={node}>
-        <Radio.Group
-          role='radiogroup'
-          required={required}
-          legend={
-            <Heading
-              level={2}
-              size='sm'
-            >
-              <Lang id={title} />
-              <RequiredIndicator required={required} />
-            </Heading>
-          }
-          description={description && <Lang id={description} />}
-          className={classes.mobileRadioGroup}
-          value={JSON.stringify(selectedRow)}
-        >
-          {data?.listItems.map((row) => (
-            <Radio
-              key={JSON.stringify(row)}
-              value={JSON.stringify(row)}
-              className={cn(classes.mobileRadio, { [classes.selectedRow]: isRowSelected(row) })}
-              onClick={() => handleRowSelect({ selectedValue: row })}
-            >
-              {tableHeadersToShowInMobile.map((key) => (
-                <div key={key}>
-                  <strong>
-                    <Lang id={tableHeaders[key]} />
-                  </strong>
-                  <span>{typeof row[key] === 'string' ? <Lang id={row[key]} /> : row[key]}</span>
-                </div>
-              ))}
-            </Radio>
-          ))}
-        </Radio.Group>
+        {saveToList ? (
```

Comment:
Letter å lese om man har en if som returnerer tidligere hvis `saveToList` eksisterer i stedet for en ternary inni her.